Java and Junit test for profiling

jorgeballesta opened the pull request:
Java logic for profiling and Junit testing used when launching the game

jorgeballesta committed:
Java and Junit test for profiling

itnifl committed:
Using EnemyTypes from difficulties

itnifl committed:
Missed this

itnifl committed:
Updating requirements

itnifl committed:
Merge branch 'TDT4250-MDD-EnemyAIAndDifficulty' into feature/EnemyProfile

itnifl commented on maze/src/main/java/main/game/maze/config/CompositionResolverImpl.java:
```diff
@@ -0,0 +1,90 @@
+  }
+  private static Map<EnemyTypes,Integer> largestRemainderAllocate(Map<EnemyTypes,Double> targets){
+    Map<EnemyTypes,Integer> floor=new EnumMap<>(EnemyTypes.class);
+    Map<EnemyTypes,Double> frac=new EnumMap<>(EnemyTypes.class);
```
Needs refactoring, hard to read

jorgeballesta committed:
Decompacted version of CompositionResolver

itnifl committed:
Moving to service

itnifl committed:
Merge branch 'TDT4250-MDD-EnemyAIAndDifficulty' into feature/EnemyProfile

itnifl committed:
Adding refactor comments

itnifl committed:
Merge branch 'TDT4250-MDD-EnemyAIAndDifficulty' into feature/EnemyProfile

jorgeballesta committed:
OpponentFactory refactoring in progress

Lots of changes made, debug and revision needed.

jorgeballesta committed:
working enemy-profiling with isolated functions for better implementation

working fine in my macos

jorgeballesta committed:
updated comments

english comments only

itnifl committed:
Logging

itnifl committed:
Merge branch 'feature/EnemyProfile' of https://github.com/itnifl/MazeGame into feature/EnemyProfile

itnifl committed:
Adding dynamic looping

itnifl committed:
Update OpponentRuntimeFactory.java

jorgeballesta committed:
working implementation (better syntax and correction + test needed)

working but need comments a testing

jorgeballesta committed:
working with new difficulty settings

New code implementation for better enemy selection + xmi rules redefined for better experience
